### PR TITLE
chore: fix vulnerabilities in attribute service

### PR DESCRIPTION
**Pull request by saxenakshitiz**
Fix vulnerabilities -

Testing /Users/kshitizsaxena/SourceCode/hypertrace/attribute-service...

Organization:      saxenakshitiz
Package manager:   gradle
Target file:       build.gradle.kts
Project name:      attribute-service
Open source:       no
Project path:      /Users/kshitizsaxena/SourceCode/hypertrace/attribute-service
Local Snyk policy: found
Licenses:          enabled

✔ Tested /Users/kshitizsaxena/SourceCode/hypertrace/attribute-service for known issues, no vulnerable paths found.

Next steps:
- Run `snyk monitor` to be notified about new related vulnerabilities.
- Run `snyk test` as part of your CI/test.

-------------------------------------------------------

Testing /Users/kshitizsaxena/SourceCode/hypertrace/attribute-service...

Organization:      saxenakshitiz
Package manager:   gradle
Target file:       build.gradle.kts
Project name:      attribute-service/attribute-projection-functions
Open source:       no
Project path:      /Users/kshitizsaxena/SourceCode/hypertrace/attribute-service
Licenses:          enabled

✔ Tested 17 dependencies for known issues, no vulnerable paths found.

Next steps:
- Run `snyk monitor` to be notified about new related vulnerabilities.
- Run `snyk test` as part of your CI/test.

-------------------------------------------------------

Testing /Users/kshitizsaxena/SourceCode/hypertrace/attribute-service...

Tested 32 dependencies for known issues, found 3 issues, 6 vulnerable paths.

Issues with no direct upgrade or patch:
  ✗ Denial of Service (DoS) [Medium Severity][https://security.snyk.io/vuln/SNYK-JAVA-COMGOOGLEPROTOBUF-3040284] in com.google.protobuf:protobuf-java@3.19.2
    introduced by org.hypertrace.core.attribute.service:attribute-service-api@0.14.17 > io.grpc:grpc-protobuf@1.44.0 > com.google.protobuf:protobuf-java@3.19.2 and 1 other path(s)
  This issue was fixed in versions: 3.16.3, 3.19.6, 3.20.3, 3.21.7
  ✗ Denial of Service (DoS) [High Severity][https://security.snyk.io/vuln/SNYK-JAVA-COMGOOGLEPROTOBUF-3167772] in com.google.protobuf:protobuf-java@3.19.2
    introduced by org.hypertrace.core.attribute.service:attribute-service-api@0.14.17 > io.grpc:grpc-protobuf@1.44.0 > com.google.protobuf:protobuf-java@3.19.2 and 1 other path(s)
  This issue was fixed in versions: 3.16.3, 3.19.6, 3.20.3, 3.21.7
  ✗ Denial of Service (DoS) [High Severity][https://security.snyk.io/vuln/SNYK-JAVA-COMGOOGLEPROTOBUF-3167774] in com.google.protobuf:protobuf-java@3.19.2
    introduced by org.hypertrace.core.attribute.service:attribute-service-api@0.14.17 > io.grpc:grpc-protobuf@1.44.0 > com.google.protobuf:protobuf-java@3.19.2 and 1 other path(s)
  This issue was fixed in versions: 3.16.3, 3.19.6, 3.20.3, 3.21.7

Organization:      saxenakshitiz
Package manager:   gradle
Target file:       build.gradle.kts
Project name:      attribute-service/attribute-projection-registry
Open source:       no
Project path:      /Users/kshitizsaxena/SourceCode/hypertrace/attribute-service
Licenses:          enabled

-------------------------------------------------------

Testing /Users/kshitizsaxena/SourceCode/hypertrace/attribute-service...

Tested 76 dependencies for known issues, found 4 issues, 8 vulnerable paths.

Issues with no direct upgrade or patch:
  ✗ Denial of Service (DoS) [Medium Severity][https://security.snyk.io/vuln/SNYK-JAVA-COMGOOGLEPROTOBUF-3040284] in com.google.protobuf:protobuf-java@3.19.2
    introduced by org.hypertrace.core.serviceframework:platform-grpc-service-framework@0.1.37 > io.grpc:grpc-services@1.47.0 > io.grpc:grpc-protobuf@1.47.0 > com.google.protobuf:protobuf-java@3.19.2 and 1 other path(s)
  This issue was fixed in versions: 3.16.3, 3.19.6, 3.20.3, 3.21.7
  ✗ Denial of Service (DoS) [High Severity][https://security.snyk.io/vuln/SNYK-JAVA-COMGOOGLEPROTOBUF-3167772] in com.google.protobuf:protobuf-java@3.19.2
    introduced by org.hypertrace.core.serviceframework:platform-grpc-service-framework@0.1.37 > io.grpc:grpc-services@1.47.0 > io.grpc:grpc-protobuf@1.47.0 > com.google.protobuf:protobuf-java@3.19.2 and 1 other path(s)
  This issue was fixed in versions: 3.16.3, 3.19.6, 3.20.3, 3.21.7
  ✗ Denial of Service (DoS) [High Severity][https://security.snyk.io/vuln/SNYK-JAVA-COMGOOGLEPROTOBUF-3167774] in com.google.protobuf:protobuf-java@3.19.2
    introduced by org.hypertrace.core.serviceframework:platform-grpc-service-framework@0.1.37 > io.grpc:grpc-services@1.47.0 > io.grpc:grpc-protobuf@1.47.0 > com.google.protobuf:protobuf-java@3.19.2 and 1 other path(s)
  This issue was fixed in versions: 3.16.3, 3.19.6, 3.20.3, 3.21.7
  ✗ Improper Certificate Validation [Medium Severity][https://security.snyk.io/vuln/SNYK-JAVA-IONETTY-1042268] in io.netty:netty-handler@4.1.79.Final
    introduced by io.grpc:grpc-netty@1.50.0 > io.netty:netty-codec-http2@4.1.79.Final > io.netty:netty-handler@4.1.79.Final and 1 other path(s)
  No upgrade or patch available

Organization:      saxenakshitiz
Package manager:   gradle
Target file:       build.gradle.kts
Project name:      attribute-service/attribute-service
Open source:       no
Project path:      /Users/kshitizsaxena/SourceCode/hypertrace/attribute-service
Licenses:          enabled

-------------------------------------------------------

Testing /Users/kshitizsaxena/SourceCode/hypertrace/attribute-service...

Tested 18 dependencies for known issues, found 3 issues, 6 vulnerable paths.

Issues to fix by upgrading:

  Upgrade io.grpc:grpc-protobuf@1.44.0 to io.grpc:grpc-protobuf@1.44.2 to fix
  ✗ Denial of Service (DoS) [Medium Severity][https://security.snyk.io/vuln/SNYK-JAVA-COMGOOGLEPROTOBUF-3040284] in com.google.protobuf:protobuf-java@3.19.2
    introduced by io.grpc:grpc-protobuf@1.44.0 > com.google.protobuf:protobuf-java@3.19.2 and 1 other path(s)
  ✗ Denial of Service (DoS) [High Severity][https://security.snyk.io/vuln/SNYK-JAVA-COMGOOGLEPROTOBUF-3167772] in com.google.protobuf:protobuf-java@3.19.2
    introduced by io.grpc:grpc-protobuf@1.44.0 > com.google.protobuf:protobuf-java@3.19.2 and 1 other path(s)
  ✗ Denial of Service (DoS) [High Severity][https://security.snyk.io/vuln/SNYK-JAVA-COMGOOGLEPROTOBUF-3167774] in com.google.protobuf:protobuf-java@3.19.2
    introduced by io.grpc:grpc-protobuf@1.44.0 > com.google.protobuf:protobuf-java@3.19.2 and 1 other path(s)

Organization:      saxenakshitiz
Package manager:   gradle
Target file:       build.gradle.kts
Project name:      attribute-service/attribute-service-api
Open source:       no
Project path:      /Users/kshitizsaxena/SourceCode/hypertrace/attribute-service
Licenses:          enabled

-------------------------------------------------------

Testing /Users/kshitizsaxena/SourceCode/hypertrace/attribute-service...

Tested 19 dependencies for known issues, found 3 issues, 6 vulnerable paths.

Issues with no direct upgrade or patch:
  ✗ Denial of Service (DoS) [Medium Severity][https://security.snyk.io/vuln/SNYK-JAVA-COMGOOGLEPROTOBUF-3040284] in com.google.protobuf:protobuf-java@3.19.2
    introduced by org.hypertrace.core.attribute.service:attribute-service-api@0.14.17 > io.grpc:grpc-protobuf@1.45.1 > com.google.protobuf:protobuf-java@3.19.2 and 1 other path(s)
  This issue was fixed in versions: 3.16.3, 3.19.6, 3.20.3, 3.21.7
  ✗ Denial of Service (DoS) [High Severity][https://security.snyk.io/vuln/SNYK-JAVA-COMGOOGLEPROTOBUF-3167772] in com.google.protobuf:protobuf-java@3.19.2
    introduced by org.hypertrace.core.attribute.service:attribute-service-api@0.14.17 > io.grpc:grpc-protobuf@1.45.1 > com.google.protobuf:protobuf-java@3.19.2 and 1 other path(s)
  This issue was fixed in versions: 3.16.3, 3.19.6, 3.20.3, 3.21.7
  ✗ Denial of Service (DoS) [High Severity][https://security.snyk.io/vuln/SNYK-JAVA-COMGOOGLEPROTOBUF-3167774] in com.google.protobuf:protobuf-java@3.19.2
    introduced by org.hypertrace.core.attribute.service:attribute-service-api@0.14.17 > io.grpc:grpc-protobuf@1.45.1 > com.google.protobuf:protobuf-java@3.19.2 and 1 other path(s)
  This issue was fixed in versions: 3.16.3, 3.19.6, 3.20.3, 3.21.7

Organization:      saxenakshitiz
Package manager:   gradle
Target file:       build.gradle.kts
Project name:      attribute-service/attribute-service-client
Open source:       no
Project path:      /Users/kshitizsaxena/SourceCode/hypertrace/attribute-service
Licenses:          enabled

-------------------------------------------------------

Testing /Users/kshitizsaxena/SourceCode/hypertrace/attribute-service...

Tested 27 dependencies for known issues, found 3 issues, 6 vulnerable paths.

Issues with no direct upgrade or patch:
  ✗ Denial of Service (DoS) [Medium Severity][https://security.snyk.io/vuln/SNYK-JAVA-COMGOOGLEPROTOBUF-3040284] in com.google.protobuf:protobuf-java@3.19.2
    introduced by org.hypertrace.core.serviceframework:platform-grpc-service-framework@0.1.37 > io.grpc:grpc-services@1.47.0 > io.grpc:grpc-protobuf@1.47.0 > com.google.protobuf:protobuf-java@3.19.2 and 1 other path(s)
  This issue was fixed in versions: 3.16.3, 3.19.6, 3.20.3, 3.21.7
  ✗ Denial of Service (DoS) [High Severity][https://security.snyk.io/vuln/SNYK-JAVA-COMGOOGLEPROTOBUF-3167772] in com.google.protobuf:protobuf-java@3.19.2
    introduced by org.hypertrace.core.serviceframework:platform-grpc-service-framework@0.1.37 > io.grpc:grpc-services@1.47.0 > io.grpc:grpc-protobuf@1.47.0 > com.google.protobuf:protobuf-java@3.19.2 and 1 other path(s)
  This issue was fixed in versions: 3.16.3, 3.19.6, 3.20.3, 3.21.7
  ✗ Denial of Service (DoS) [High Severity][https://security.snyk.io/vuln/SNYK-JAVA-COMGOOGLEPROTOBUF-3167774] in com.google.protobuf:protobuf-java@3.19.2
    introduced by org.hypertrace.core.serviceframework:platform-grpc-service-framework@0.1.37 > io.grpc:grpc-services@1.47.0 > io.grpc:grpc-protobuf@1.47.0 > com.google.protobuf:protobuf-java@3.19.2 and 1 other path(s)
  This issue was fixed in versions: 3.16.3, 3.19.6, 3.20.3, 3.21.7

Organization:      saxenakshitiz
Package manager:   gradle
Target file:       build.gradle.kts
Project name:      attribute-service/attribute-service-factory
Open source:       no
Project path:      /Users/kshitizsaxena/SourceCode/hypertrace/attribute-service
Licenses:          enabled

-------------------------------------------------------

Testing /Users/kshitizsaxena/SourceCode/hypertrace/attribute-service...

Tested 52 dependencies for known issues, found 3 issues, 3 vulnerable paths.

Issues to fix by upgrading:

  Upgrade com.fasterxml.jackson.core:jackson-databind@2.13.2.2 to com.fasterxml.jackson.core:jackson-databind@2.13.4.2 to fix
  ✗ Denial of Service (DoS) [Medium Severity][https://security.snyk.io/vuln/SNYK-JAVA-COMFASTERXMLJACKSONCORE-3038426] in com.fasterxml.jackson.core:jackson-databind@2.13.2.2
    introduced by com.fasterxml.jackson.core:jackson-databind@2.13.2.2
  ✗ Denial of Service (DoS) [Medium Severity][https://security.snyk.io/vuln/SNYK-JAVA-COMFASTERXMLJACKSONCORE-3038424] in com.fasterxml.jackson.core:jackson-databind@2.13.2.2
    introduced by com.fasterxml.jackson.core:jackson-databind@2.13.2.2

  Upgrade com.google.protobuf:protobuf-java-util@3.19.2 to com.google.protobuf:protobuf-java-util@3.20.0 to fix
  ✗ Deserialization of Untrusted Data [Medium Severity][https://security.snyk.io/vuln/SNYK-JAVA-COMGOOGLECODEGSON-1730327] in com.google.code.gson:gson@2.8.6
    introduced by com.google.protobuf:protobuf-java-util@3.19.2 > com.google.code.gson:gson@2.8.6

Organization:      saxenakshitiz
Package manager:   gradle
Target file:       build.gradle.kts
Project name:      attribute-service/attribute-service-impl
Open source:       no
Project path:      /Users/kshitizsaxena/SourceCode/hypertrace/attribute-service
Licenses:          enabled

-------------------------------------------------------

Testing /Users/kshitizsaxena/SourceCode/hypertrace/attribute-service...

Organization:      saxenakshitiz
Package manager:   gradle
Target file:       build.gradle.kts
Project name:      attribute-service/attribute-service-tenant-api
Open source:       no
Project path:      /Users/kshitizsaxena/SourceCode/hypertrace/attribute-service
Licenses:          enabled

✔ Tested 15 dependencies for known issues, no vulnerable paths found.

Next steps:
- Run `snyk monitor` to be notified about new related vulnerabilities.
- Run `snyk test` as part of your CI/test.

-------------------------------------------------------

Testing /Users/kshitizsaxena/SourceCode/hypertrace/attribute-service...

Organization:      saxenakshitiz
Package manager:   gradle
Target file:       build.gradle.kts
Project name:      attribute-service/caching-attribute-service-client
Open source:       no
Project path:      /Users/kshitizsaxena/SourceCode/hypertrace/attribute-service
Licenses:          enabled

✔ Tested 47 dependencies for known issues, no vulnerable paths found.

Next steps:
- Run `snyk monitor` to be notified about new related vulnerabilities.
- Run `snyk test` as part of your CI/test.

Tested 10 projects, 6 contained vulnerable paths.

### CODE REVIEW

File: attribute-service-api/build.gradle.kts [hunk 1]
```diff
@@ -15,7 +15,7 @@ val generateLocalGoGrpcFiles = false
 
 protobuf {
   protoc {
-    artifact = "com.google.protobuf:protoc:3.15.6"
+    artifact = "com.google.protobuf:protoc:3.21.1"
```
Comment (kotharironak) [inline attribute-service-api/build.gradle.kts:18]: shall we move to https://mvnrepository.com/artifact/com.google.protobuf/protobuf-java/3.22.0 ?

Comment (GurtejSohi) [inline attribute-service-api/build.gradle.kts:18]: we should keep the versions same for the modules coming from the same group - https://github.com/hypertrace/attribute-service/pull/160/files#diff-fd6def69030cceecee0ba666b24df9af4eb511efc9c8b849c0bc168a1596b812R17
```suggestion
    artifact = "com.google.protobuf:protoc:3.21.12"
```

Comment (saxenakshitiz) [reply attribute-service-api/build.gradle.kts:18]: will require code change. Will it be okay to stick to 3.21.12?

Comment (saxenakshitiz) [reply attribute-service-api/build.gradle.kts]: Done

**Review by GurtejSohi — APPROVED** [review]

**Pull request merged**